rust/kernel/net/ethernet.rs

```diff
@@ -0,0 +1,435 @@
+            bindings::ETH_P_8021AH => Ok(Self::_8021ah),
+            bindings::ETH_P_MVRP => Ok(Self::Mvrp),
+            bindings::ETH_P_1588 => Ok(Self::_1588),
+            bindings::ETH_P_NCSI => Ok(Self::Ncsi),
+            bindings::ETH_P_PRP => Ok(Self::Prp),
+            bindings::ETH_P_CFM => Ok(Self::Cfm),
+            bindings::ETH_P_FCOE => Ok(Self::FcoE),
+            bindings::ETH_P_IBOE => Ok(Self::IboE),
+            bindings::ETH_P_TDLS => Ok(Self::Tdls),
+            bindings::ETH_P_FIP => Ok(Self::Fip),
+            bindings::ETH_P_80221 => Ok(Self::_80221),
+            bindings::ETH_P_HSR => Ok(Self::Hsr),
+            bindings::ETH_P_NSH => Ok(Self::Nsh),
+            bindings::ETH_P_LOOPBACK => Ok(Self::Loopback),
+            bindings::ETH_P_QINQ1 => Ok(Self::QinQ1),
+            bindings::ETH_P_QINQ2 => Ok(Self::QinQ2),
+            bindings::ETH_P_QINQ3 => Ok(Self::QinQ3),
+            bindings::ETH_P_EDSA => Ok(Self::Edsa),
+            bindings::ETH_P_DSA_8021Q => Ok(Self::Dsa8021Q),
+            bindings::ETH_P_DSA_A5PSW => Ok(Self::DsaA5Psw),
+            bindings::ETH_P_IFE => Ok(Self::Ife),
+            bindings::ETH_P_AF_IUCV => Ok(Self::AfIucv),
+            bindings::ETH_P_802_3_MIN => Ok(Self::Min8023),
+            bindings::ETH_P_802_3 => Ok(Self::_8023),
+            bindings::ETH_P_AX25 => Ok(Self::Ax25),
+            bindings::ETH_P_ALL => Ok(Self::All),
+            bindings::ETH_P_802_2 => Ok(Self::_8022),
+            bindings::ETH_P_SNAP => Ok(Self::Snap),
+            bindings::ETH_P_DDCMP => Ok(Self::DDcmp),
+            bindings::ETH_P_WAN_PPP => Ok(Self::WanPpp),
+            bindings::ETH_P_PPP_MP => Ok(Self::PppMp),
+            bindings::ETH_P_LOCALTALK => Ok(Self::LocalTalk),
+            bindings::ETH_P_CAN => Ok(Self::Can),
+            bindings::ETH_P_CANFD => Ok(Self::CanFd),
+            bindings::ETH_P_CANXL => Ok(Self::CanXl),
+            bindings::ETH_P_PPPTALK => Ok(Self::PppTalk),
+            bindings::ETH_P_TR_802_2 => Ok(Self::Tr8022),
+            bindings::ETH_P_MOBITEX => Ok(Self::Mobitex),
+            bindings::ETH_P_CONTROL => Ok(Self::Control),
+            bindings::ETH_P_IRDA => Ok(Self::Irda),
+            bindings::ETH_P_ECONET => Ok(Self::Econet),
+            bindings::ETH_P_HDLC => Ok(Self::Hdlc),
+            bindings::ETH_P_ARCNET => Ok(Self::ArcNet),
+            bindings::ETH_P_DSA => Ok(Self::Dsa),
+            bindings::ETH_P_TRAILER => Ok(Self::Trailer),
+            bindings::ETH_P_PHONET => Ok(Self::Phonet),
+            bindings::ETH_P_IEEE802154 => Ok(Self::IEEE802154),
+            bindings::ETH_P_CAIF => Ok(Self::Caif),
+            bindings::ETH_P_XDSA => Ok(Self::XDsa),
+            bindings::ETH_P_MAP => Ok(Self::Map),
+            bindings::ETH_P_MCTP => Ok(Self::Mctp),
+            _ => Err(UnknownProtoError(u)),
+        }
+    }
+}
+
+/// Ethernet Address Type
+pub type Address = [u8; Lengths::Address as usize];
```

Comment:
Maybe use a newtype? `pub struct Address(pub [u8; bindings::ETH_ALEN as usize]);`

Reply:
(Done in the latest force push)